sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -416,12 +424,15 @@ case class ShuffledHashJoinExec(
          |}
        """.stripMargin)
 
+    val isFullOuterJoin = joinType == FullOuter
     val joinWithUniqueKey = codegenFullOuterJoinWithUniqueKey(
```

Comment:
These helper method names are tied to full outer join. Maybe it is better to revise them. Otherwise code readers might be confused.